tree/tree/src/TBranchElement.cxx

```diff
@@ -2248,7 +2254,60 @@ void TBranchElement::InitInfo()
                currentVersion = info->GetClassVersion();
             }
 
-            for (Int_t i = 0; i < nbranches; ++i) {
+            // First find the first branch corresponding to the same class as 'this'
+            // branch
+            Int_t index = branches->IndexOf(this);
+            Int_t firstindex = -1;
+            Int_t lastindex = nbranches - 1;
+            if (index > 0) {
```

Comment:
Can `index <= 0` happen? Is this worth a `Warning`?

Reply:
Yes, all the time.  It is the case of a toplevel branch.

Reply:
But there was a problem!  The case `index == 0` was incorrectly handled.
```suggestion
            if (index >= 0) {
```